ddtrace/contrib/internal/langchain/utils.py

```diff
@@ -0,0 +1,79 @@
+import inspect
+import sys
+
+
+class BaseTracedLangChainStreamResponse:
+    def __init__(self, generator, integration, span, on_span_finish):
+        self._generator = generator
+        self._dd_integration = integration
+        self._dd_span = span
+        self._on_span_finish = on_span_finish
+        self._chunks = []
+
+
+class TracedLangchainStreamResponse(BaseTracedLangChainStreamResponse):
+    def __iter__(self):
+        try:
+            for chunk in self._generator.__iter__():
+                self._chunks.append(chunk)
+                yield chunk
+        except Exception:
+            self._dd_span.set_exc_info(*sys.exc_info())
+            self._dd_integration.metric(self._dd_span, "incr", "request.error", 1)
+            raise
+        finally:
+            self._on_span_finish(self._dd_span, self._chunks, error=bool(self._dd_span.error))
+            self._dd_span.finish()
+
+
+class TracedLangchainAsyncStreamResponse(BaseTracedLangChainStreamResponse):
+    async def __aiter__(self):
+        try:
+            async for chunk in self._generator.__aiter__():
+                self._chunks.append(chunk)
+                yield chunk
+        except Exception:
+            self._dd_span.set_exc_info(*sys.exc_info())
+            self._dd_integration.metric(self._dd_span, "incr", "request.error", 1)
+            raise
+        finally:
+            self._on_span_finish(self._dd_span, self._chunks, error=bool(self._dd_span.error))
```

Comment:
This is part of a larger refactor I'm pushing for, but can we get rid of the `error` arg in these handlers? This is already captured as part of `span.error` so I don't see why we need this to be passed explicitly

Reply:
yep can do